### ISSUE TITLE
False positive for unsafe comparison of hash that are susceptible to timing attack

### ISSUE THREAD

**Issue by titicalvados** [find-sec-bugs/find-sec-bugs#558]
## Environment

Java 11, Maven, SonarQube 6.7.7 (LTS) with findsecbugs latest plugin 

## Problem

Rule findsecbugs:UNSAFE_HASH_EQUALS
Reference CWE-203: Information Exposure Through DiscrepancyKey
(https://cwe.mitre.org/data/definitions/203.html 2)

findsecbugs considers some strings are hash and report an unsafe comparison of hash that are susceptible to timing attack whereas it is some randomly generated strings (UUID.randomUUID().toString())

I can’t explain why the shareId string is considered as a hash.
Any idea ?
Regards

exemple of code generating the issue:
## Code (If needed)

```java
//if no share match shareId return exception
if (foundFile.getShares().stream().noneMatch(a -> a.getShareId().equals(shareId))) {
		throw new ShareUnknownException(String.format("Unknown share identifier: %s", shareId));
}
```
Screenshot :

![SonarQube](https://user-images.githubusercontent.com/50456999/85122996-c06c1980-b227-11ea-8c97-bc2128d4f0c0.png)

**Comment by h3xstream** [comment]
The false positive occurs because the keyword SHA is associate to the hash algorithm.
https://github.com/find-sec-bugs/find-sec-bugs/blob/master/findsecbugs-plugin/src/main/java/com/h3xstream/findsecbugs/password/HashUnsafeEqualsDetector.java#L64

I could add "shared" in whitelist for this detector as it is a common keyword.

**Comment by h3xstream** [comment]
This issue was tagged [wontfix] by mistake.

**Comment by h3xstream** [comment]
Fix in #565
